### PR TITLE
W-14270458  Test Framework does not handle some input values correctly - part2

### PR DESCRIPTION
**Pull request by bairenlong-sfdc**

### CODE REVIEW

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaFieldDefinition.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ public Object createObjectWithGivenValue(String fieldValue){
         }
 
         if (this.dataType.isNumber()) {
-            if ("NULL".equalsIgnoreCase(fieldValue)) {
+            if ("NULL".equalsIgnoreCase(fieldValue.trim())) {
```
Comment (bairenlong-sfdc) [inline test-utils/src/main/java/com/force/formula/v2/data/FormulaFieldDefinition.java:100]: Need to handle input value like "  NULL  "

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java [hunk 2]
```diff
@@ -41,17 +41,25 @@ public FormulaTestData(String testCase, String input, List<String> outputs, List
         //To make sure that split does not ignore the last empty input, we added a -1 in the following method call
         String[] inputArray = input.trim().split(",",-1);
 
-        Validate.isTrue((inputFields.size()==0)||((inputFields.size()>0) && (inputArray.length==inputFields.size())),"Each field needs to have a " +
```
Comment (bairenlong-sfdc) [inline test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java:44]: Currently, it is impossible to know which test case is wrong.

**Review by jonathonmar — APPROVED** [review]

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java [hunk 3]
```diff
@@ -41,17 +41,25 @@ public FormulaTestData(String testCase, String input, List<String> outputs, List
...
-                "corresponding input value defined. Input Fields Size: " + inputFields.size()+" , Input Values Size: "
-                + inputArray.length);
-        Validate.isTrue(executionPaths.size()==outputs.size(),"Each executionPath needs to have " +
-                "an expectedOutput in order in which the execution paths are indexed or defined in the testcase");
+        Validate.isTrue((inputFields.size()==0)||((inputFields.size()>0) && (inputArray.length==inputFields.size())),
+                        "Each field needs to have a corresponding input value defined. Input Fields Size: " +
```
Comment (jonathonmar) [inline test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java:45]: Looks like "Input Fields Size: " should be "Testcase: " here.

File: test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java (hunk 2 above)
Comment (bairenlong-sfdc) [reply test-utils/src/main/java/com/force/formula/v2/data/FormulaTestData.java:44]: yes, will update it next time.

**Pull request merged**